node/sync.go

```diff
@@ -321,8 +321,12 @@ func (d *Pegnetd) ApplyTransactionBatchesInHolding(ctx context.Context, sqlTx *s
 			}
 
 			err = d.applyTransactionBatch(sqlTx, txBatch, rates, currentHeight)
-			if err != nil && err != pegnet.InsufficientBalanceErr {
-				return nil
+			if err != nil && err != pegnet.InsufficientBalanceErr && err != pegnet.PFCTOneWayError {
+				return err
+			} else if err == pegnet.InsufficientBalanceErr {
+				d.Pegnet.SetTransactionHistoryExecuted(sqlTx, txBatch, -1)
+			} else if err == pegnet.PFCTOneWayError {
+				d.Pegnet.SetTransactionHistoryExecuted(sqlTx, txBatch, -2)
```

Comment:
why -2 instead of also just -1?

Reply:
Right now -1 is insufficient balance. Carl wants the explorer to explain why the tx was rejected, so I figure `<0` means rejected, and the exact number is the error code that indicates why

Reply:
probably doesn't hurt